If mypy fails it should report the error back to lintrunner

aorenste opened the pull request:
This happened to me because I had a bad LD_LIBRARY_PATH and mypy was failing to run (.so load error) - but lintrunner was silent about the underlying problem.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #145554
* #145553
* #145552
* #145551
* __->__ #145550

Differential Revision: [D68593081](https://our.internmc.facebook.com/intern/diff/D68593081)

aorenste committed:
If mypy fails it should report the error back to lintrunner

[ghstack-poisoned]

aorenste committed:
Update on "If mypy fails it should report the error back to lintrunner"




[ghstack-poisoned]

aorenste commented:
@aorenste has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D68593081).

bobrenjc93 reviewed (approved):
Well that makes sense :)

aorenste committed:
Update on "If mypy fails it should report the error back to lintrunner"


Differential Revision: [D68593081](https://our.internmc.facebook.com/intern/diff/D68593081)

[ghstack-poisoned]

aorenste commented:
@aorenste has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D68593081).

aorenste commented:
@pytorchbot merge
